bycycle/features/shape.py

```diff
@@ -0,0 +1,213 @@
+    shape_features['time_trough'] = rises - decays[:-1]
+
+    # Determine extrema voltage
+    shape_features['volt_peak'] = sig[peaks[1:]]
+    shape_features['volt_trough'] = sig[troughs[:-1]]
+
+    # Determine rise and decay characteristics
+    shape_features['time_decay'] = (troughs[1:] - peaks[1:])
+    shape_features['time_rise'] = (peaks[1:] - troughs[:-1])
+
+    shape_features['volt_decay'] = sig[peaks[1:]] - sig[troughs[1:]]
+    shape_features['volt_rise'] = sig[peaks[1:]] - sig[troughs[:-1]]
+    shape_features['volt_amp'] = (shape_features['volt_decay'] + shape_features['volt_rise']) / 2
+
+    # Compute rise-decay symmetry features
+    shape_features['time_rdsym'] = shape_features['time_rise'] / shape_features['period']
+
+    # Compute peak-trough symmetry features
+    shape_features['time_ptsym'] = shape_features['time_peak'] / \
+            (shape_features['time_peak'] + shape_features['time_trough'])
+
+    # Compute average oscillatory amplitude estimate during cycle
+    amp = amp_by_time(sig, fs, f_range, hilbert_increase_n=hilbert_increase_n, n_cycles=3)
+    shape_features['band_amp'] = [np.mean(amp[troughs[sig_idx]:troughs[sig_idx + 1]]) for sig_idx in
+                                  range(len(df_samples['sample_peak']))]
+
+    # Convert feature dictionary into a DataFrame
+    df_features = pd.DataFrame.from_dict(shape_features)
+
+    # Rename columns if they are actually trough-centered
+    if center_extrema == 'trough':
+
+        samples_rename_dict = {'sample_peak': 'sample_trough',
+                               'sample_zerox_decay': 'sample_zerox_rise',
+                               'sample_zerox_rise': 'sample_zerox_decay',
+                               'sample_last_trough': 'sample_last_peak',
+                               'sample_next_trough': 'sample_next_peak'}
+
+        features_rename_dict = {'time_peak': 'time_trough',
+                                'time_trough': 'time_peak',
+                                'volt_peak': 'volt_trough',
+                                'volt_trough': 'volt_peak',
+                                'time_rise': 'time_decay',
+                                'time_decay': 'time_rise',
+                                'volt_rise': 'volt_decay',
+                                'volt_decay': 'volt_rise'}
+
+        df_samples.rename(columns=samples_rename_dict, inplace=True)
+        df_features.rename(columns=features_rename_dict, inplace=True)
+
+        # Need to reverse symmetry measures
+        df_features['volt_peak'] = -df_features['volt_peak']
+        df_features['volt_trough'] = -df_features['volt_trough']
+        df_features['time_rdsym'] = 1 - df_features['time_rdsym']
+        df_features['time_ptsym'] = 1 - df_features['time_ptsym']
+
+    if return_samples:
+        return df_features, df_samples
+
+    return df_features
+
+
+def compute_samples(peaks, troughs, decays, rises):
```

Comment:
This function sounds like it should compute all the cyclepoints - but it seems like it mostly just collects them. 

Any reason to not make this function do the computing as well? As it gets called from the above function, then the following lines could be adopted into this function:

```
# Find extrema and zero-crossings locations in the signal
peaks, troughs = find_extrema(sig, fs, f_range, **find_extrema_kwargs)
rises, decays = find_zerox(sig, peaks, troughs)
```

The API would be updated to it to work accordingly, to something like:
`def compute_samples(sig, fs, f_range, **extram_kwargs)`

Reply:
This is a good idea, updated (da3fd5c).